### PR TITLE
Resynchronize Slice Compiler

### PR DESCRIPTION
**Pull request by InsertCreativityHere**
This PR resynchronizes the Bison+Flex files with their generated code.

It also fixes their formatting, since `clang-format` is incapable of formatting them.

Additionally, it updates the warnings we ignore. Some of them are now unnecssary, but for some reason,
there's some new warnings now that needed to be ignored for the builds to work.

### CODE REVIEW

File: cpp/src/Slice/Scanner.l [hunk 1]
```diff
@@ -552,154 +548,150 @@ floating_literal    (({fractional_constant}{exponent_part}?)|((\+|-)?{dec}+{expo
...
-        if(pos->first != id)
+        const StringTokenMap::const_iterator pos = keywordMap.find(id);
+        if (pos != keywordMap.end())
         {
-            currentUnit->error("illegal identifier: `" + id + "' differs from keyword `" + pos->first +
-                        "' only in capitalization");
-            id = pos->first;
+            if (pos->first != id)
+            {
+                currentUnit->error("illegal identifier: `" + id + "' differs from keyword `" + pos->first +
```
Comment (pepone) [inline cpp/src/Slice/Scanner.l:562]: minor: this is aligned differently in the generated version.

File: cpp/src/Slice/Scanner.l [hunk 2]
```diff
@@ -552,154 +548,150 @@ floating_literal    (({fractional_constant}{exponent_part}?)|((\+|-)?{dec}+{expo
...
+        keywordMap["string"] = ICE_STRING;
+        keywordMap["Object"] = ICE_OBJECT;
+        keywordMap["const"] = ICE_CONST;
+        keywordMap["false"] = ICE_FALSE;
+        keywordMap["true"] = ICE_TRUE;
+        keywordMap["idempotent"] = ICE_IDEMPOTENT;
+        keywordMap["tag"] = ICE_TAG;
+        // 'optional' is kept as an alias for 'tag' for backwards compatability.
+        // We need a separate token type since we infer 'optional T' to mean 'tag T?'.
+        // But for 'tag' we require an optional (nullable) type. No inferencing is done.
```
Comment (pepone) [inline cpp/src/Slice/Scanner.l:680]: Not related to the PR, but we need to rework this comment, I think it comes from the 4.0 merge. We don't have nullable types in Ice 3.8.

**Review by pepone — APPROVED** [review]

File: cpp/src/Slice/Scanner.l (hunk 1 above)
Comment (InsertCreativityHere) [reply cpp/src/Slice/Scanner.l]: Good catch and fair enough!
Fixed!

File: cpp/src/Slice/Scanner.l (hunk 2 above)
Comment (InsertCreativityHere) [reply cpp/src/Slice/Scanner.l]: Yes, I just went ahead and fixed it (by removing the last two sentences).
Might as well do it now.

**Pull request merged**